golem-test-framework/src/dsl/mod.rs

```diff
@@ -82,15 +82,15 @@ pub trait TestDsl {
     async fn get_worker_metadata(
         &self,
         worker_id: &WorkerId,
-    ) -> crate::Result<Option<WorkerMetadata>>;
+    ) -> crate::Result<Option<(WorkerMetadata, Option<String>)>>;
```

Comment:
can we create a struct for this tuple or make it part of the WorkerMetadata?

Reply:
It is intentionally not part of the worker metadata (because that's the actual serialized struct that is stored in redis, and the error information is gathered from oplog entries) .

I don't like using tuples anywhere in general, but in these tests, I felt that having 
``` let (metadata, _) = ...` where we don't care about the last failure is less annoying than completely changing the result type to something that wraps both. 

But I'm open to change my mind :)